Vaadin 14 starter Spring Boot doesn't work

Hi,

I downloaded the starter Spring Boot project from the Vaadin website and tried to get it working. The result is this in the browser: 

```
Webpack Error
i ｢wds｣: Project is running at http://localhost:51352/webpack-dev-server/
i ｢wds｣: webpack output is served from /
i ｢wds｣: Content not from webpack is served from C:\Users\taaghal2\Documents\Code\my-starter-project\target\classes\META-INF\VAADIN, src/main/webapp
Watchdog connected.
         Serving the 'stats.json' file dynamically.
× ｢wdm｣: Hash: fd55b36c03c62e513362
Version: webpack 4.30.0
Time: 2212ms
Built at: 2019-11-21 11:40:49
                                                                     Asset       Size      Chunks                    Chunk Names
                         build/vaadin-bundle-e5ac83959d1048f9b115.cache.js   1.83 MiB      bundle  [emitted]         bundle
                      build/vaadin-bundle-e5ac83959d1048f9b115.cache.js.gz    495 KiB              [emitted]         
                     build/vaadin-bundle.es5-f31025dbf8ff0d1a6a8e.cache.js   2.04 MiB  bundle.es5  [emitted]  [big]  bundle.es5
                  build/vaadin-bundle.es5-f31025dbf8ff0d1a6a8e.cache.js.gz    524 KiB              [emitted]         
                                          build\webcomponentsjs\LICENSE.md   1.52 KiB              [emitted]         
                                           build\webcomponentsjs\README.md   10.5 KiB              [emitted]         
                         build\webcomponentsjs\bundles\webcomponents-ce.js   19.5 KiB              [emitted]         
                     build\webcomponentsjs\bundles\webcomponents-ce.js.map    129 KiB              [emitted]         
                   build\webcomponentsjs\bundles\webcomponents-sd-ce-pf.js    113 KiB              [emitted]         
               build\webcomponentsjs\bundles\webcomponents-sd-ce-pf.js.map    675 KiB              [emitted]         
                      build\webcomponentsjs\bundles\webcomponents-sd-ce.js   84.8 KiB              [emitted]         
                  build\webcomponentsjs\bundles\webcomponents-sd-ce.js.map    548 KiB              [emitted]         
                         build\webcomponentsjs\bundles\webcomponents-sd.js   66.2 KiB              [emitted]         
                     build\webcomponentsjs\bundles\webcomponents-sd.js.map    419 KiB              [emitted]         
                      build\webcomponentsjs\custom-elements-es5-adapter.js  950 bytes              [emitted]         
                                        build\webcomponentsjs\package.json    2.4 KiB              [emitted]         
build\webcomponentsjs\src\entrypoints\custom-elements-es5-adapter-index.js  659 bytes              [emitted]         
       build\webcomponentsjs\src\entrypoints\webcomponents-bundle-index.js   1.66 KiB              [emitted]         
           build\webcomponentsjs\src\entrypoints\webcomponents-ce-index.js  675 bytes              [emitted]         
        build\webcomponentsjs\src\entrypoints\webcomponents-sd-ce-index.js  860 bytes              [emitted]         
     build\webcomponentsjs\src\entrypoints\webcomponents-sd-ce-pf-index.js   1.22 KiB              [emitted]         
           build\webcomponentsjs\src\entrypoints\webcomponents-sd-index.js  767 bytes              [emitted]         
                             build\webcomponentsjs\webcomponents-bundle.js    114 KiB              [emitted]         
                         build\webcomponentsjs\webcomponents-bundle.js.map    677 KiB              [emitted]         
                             build\webcomponentsjs\webcomponents-loader.js   6.13 KiB              [emitted]         
Entrypoint bundle = build/vaadin-bundle-e5ac83959d1048f9b115.cache.js
Entrypoint bundle.es5 [big] = build/vaadin-bundle.es5-f31025dbf8ff0d1a6a8e.cache.js
[../node_modules/@vaadin/vaadin-button/theme/lumo/vaadin-button.js] 72 bytes {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-button/theme/lumo/vaadin-button.js] 72 bytes {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/color.js] 7.63 KiB {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/color.js] 7.43 KiB {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/icons.js] 87 bytes {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/icons.js] 87 bytes {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/sizing.js] 679 bytes {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/sizing.js] 664 bytes {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/spacing.js] 1.29 KiB {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/spacing.js] 1.27 KiB {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/style.js] 1.24 KiB {bundle.es5} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/style.js] 1.23 KiB {bundle} [built]
[../node_modules/@vaadin/vaadin-lumo-styles/typography.js] 3.61 KiB {bundle.es5} [built]
[../target/frontend/generated-flow-imports.js] 770 bytes {bundle.es5} [built]
[../target/frontend/generated-flow-imports.js] 772 bytes {bundle} [built]
    + 173 hidden modules

ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/flow-component-renderer.js' in 'C:\Users\taaghal2\Documents\Code\my-starter-project\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 12:0-58

ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/flow-component-renderer.js' in 'C:\Users\taaghal2\Documents\Code\my-starter-project\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 12:0-58
i ｢wdm｣: Failed to compile.
```

The only changes I made were in the POM file, adding the dependency for compatibility mode, and enabling the `frontend-maven-plugin` for automatically installing node and npm. The error remains unchanged if I remove the compatibility dependency. OpenJDK 1.8.0_232 on Windows 10, in case it matters.

Thanks

--- a/flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/PrepareFrontendMojo.java
+++ b/flow-maven-plugin/src/main/java/com/vaadin/flow/plugin/maven/PrepareFrontendMojo.java
@@ -19,8 +19,11 @@ import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 import org.apache.commons.io.FileUtils;
+import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugin.logging.Log;
@@ -147,14 +150,23 @@ public class PrepareFrontendMojo extends FlowModeAbstractMojo {
                             + "'. Verify that you may write to path.", e);
         }
         try {
-            new NodeTasks.Builder(getClassFinder(project), npmFolder,
+            NodeTasks.Builder builder = new NodeTasks.Builder(getClassFinder(project), npmFolder,
                     generatedFolder, frontendDirectory)
                             .withWebpack(webpackOutputDirectory,
                                     webpackTemplate, webpackGeneratedTemplate)
                             .createMissingPackageJson(true)
                             .enableImportsUpdate(false)
-                            .enablePackagesUpdate(false).runNpmInstall(false)
-                            .build().execute();
+                            .enablePackagesUpdate(false).runNpmInstall(false);
+            // If building a jar project copy jar artifact contents now as we might
+            // not be able to read files from jar path.
+            if("jar".equals(project.getPackaging())) {
+                Set<File> jarFiles = project.getArtifacts().stream()
+                        .filter(artifact -> "jar".equals(artifact.getType()))
+                        .map(Artifact::getFile).collect(Collectors.toSet());
+                builder.copyResources(jarFiles);
+            }
+
+            builder.build().execute();
         } catch (ExecutionFailedException exception) {
             throw new MojoFailureException(
                     "Could not execute prepare-frontend goal.", exception);

--- a/flow-maven-plugin/src/test/java/com/vaadin/flow/plugin/maven/PrepareFrontendMojoTest.java
+++ b/flow-maven-plugin/src/test/java/com/vaadin/flow/plugin/maven/PrepareFrontendMojoTest.java
@@ -24,7 +24,6 @@ import com.vaadin.flow.plugin.TestUtils;
 import elemental.json.Json;
 import elemental.json.JsonObject;
 import elemental.json.impl.JsonUtil;
-
 import static com.vaadin.flow.plugin.maven.BuildFrontendMojoTest.assertContainsPackage;
 import static com.vaadin.flow.plugin.maven.BuildFrontendMojoTest.getPackageJson;
 import static com.vaadin.flow.plugin.maven.BuildFrontendMojoTest.setProject;
@@ -54,6 +53,7 @@ public class PrepareFrontendMojoTest {
     private File projectBase;
     private File webpackOutputDirectory;
     private File tokenFile;
+    private MavenProject project;
 
     @Before
     public void setup() throws Exception {
@@ -62,7 +62,7 @@ public class PrepareFrontendMojoTest {
         tokenFile = new File(temporaryFolder.getRoot(),
                 VAADIN_SERVLET_RESOURCES + TOKEN_FILE);
 
-        MavenProject project = Mockito.mock(MavenProject.class);
+        project = Mockito.mock(MavenProject.class);
         Mockito.when(project.getBasedir()).thenReturn(projectBase);
 
         nodeModulesPath = new File(projectBase, NODE_MODULES);
@@ -160,6 +160,19 @@ public class PrepareFrontendMojoTest {
         JsonObject packageJsonObject = getPackageJson(packageJson);
         assertContainsPackage(packageJsonObject.getObject("dependencies"),
                 "foo");
+    }
+
+    @Test
+    public void jarPackaging_copyProjectFrontendResources()
+            throws MojoExecutionException, MojoFailureException,
+            IllegalAccessException {
+        Mockito.when(project.getPackaging()).thenReturn("jar");
+
+        ReflectionUtils.setVariableValueInObject(mojo, "project", project);
+
+        mojo.execute();
+
+        Mockito.verify(project, Mockito.atLeastOnce()).getArtifacts();
     }
 
     private void assertPackageJsonContent() throws IOException {

